src/core/server/saved_objects/saved_objects_service.test.ts

```diff
@@ -307,9 +307,12 @@ describe('SavedObjectsService', () => {
         warningNodes: [],
         kibanaVersion: '8.0.0',
       });
-      setImmediate(() => {
-        expect(migratorInstanceMock.runMigrations).toHaveBeenCalledTimes(1);
-        done();
+
+      return new Promise<void>((done) => {
+        setImmediate(() => {
```

Comment:
The `timers/promises` module has a promisified version of `setImmediate` you could use https://nodejs.org/api/timers.html#timerspromisessetimmediatevalue-options